srcpkgs/rtl8192eu-dkms/template

```diff
@@ -0,0 +1,28 @@
+# Template file for 'rtl8192eu-dkms'
+pkgname=rtl8192eu-dkms
+version=20221031
+revision=1
+_modver=2022.10.31
+_gitrev=982f7907c478724b9fc4febd476b38c72dd4c1f2
+archs='*'
+wrksrc="rtl8192eu-linux-driver-${_gitrev}"
+depends="dkms bc"
+short_desc="Realtek 8192EU USB WiFi driver (DKMS)"
+maintainer="Phicem <phicem@gmx.com>"
+license="GPL-2.0-only"
```

Comment:
> The original code is copyrighted, but I don't know by whom. The driver download does not contain license information; please open an issue if you are the copyright holder.

That doesn't look like GPL2

Reply:
I think it's not as bad as it looks. 

The comment that you quoted above refers to the original code (first commit of the repo), which is downloadable from DLINK website. I tried to get into depth into what was released under GPL and what was not.

### In the original code
- There is no general license information
- the Makefile (and some other trivial config files and trivial scripts) have no author and no license information
- most of the 333 source files (.h and .c) have license information (GPL2), except 39 files
- there are 16 .c files without license information:
    - 2 files have no author information (`./core/rtw_wapi.c` and `./core/rtw_wapi_sms4.c`)
    - 12 files have an author name ("Cosa") (`./hal/OUTSRC-BTCoexist/HalBtc*`)
    - 2 files are Copyright Realtek (`./hal/rtl8192e/Hal8192EPwrSeq.c` and `./os_dep/linux/wifi_regd.c`)
- there are 23 header files without license information.

### In current code
Recent versions of the repo (which would be what would be packaged here) have even less license-less files:
- the Makefile (and some other trivial config files and trivial scripts) still have no author or license information
- all but 4 of the 512 source files (.h and .c) have license information (GPLv2)
- there is 1 .c file without license information:
   - this file is `./hal/btc/halbtc8192e2ant.c`
   - this file mentions an author name ("Cosa") in the history section
   - this file exists in a different form in many other repositories, including the linux kernel itself, with GPL license. What I mean by 'different form' is that the code has undergone many changes, but the mentioned author is the same. See for instance: [in the linux kernel](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/net/wireless/realtek/rtlwifi/btcoexist/halbtc8192e2ant.c?h=v6.1-rc3&id=f1d2b4d338bf090296f67830025228872fd52513)
- there are 3 header files without author or license information
    - one is a trivial file `./include/rtw_version.h`
    - the second one is `./hal/btc/halbtc8192e2ant.h` and also exists in the linux kernel in another form (many changes in the code but same base)
    - the thirst one is `./hal/phydm/phydm_regtable.h` but is a long list of trivial `#define`

### Conclusion
Apart from the makefile and some trivial scripts and headers, there are only two files that are not explicitely GPLed : `./hal/btc/halbtc8192e2ant.c` and its corresponding header. The fact that modified versions of these files can be found as GPL-released files tends to reassure me, but still, I have no proof that the exact files we use are GPL. Moreover, I have no license information at all about the makefile.

If we say that this is a problem, could we put the package in the `nonfree` repository, with a license field like `license="GPL-2.0-only, custom:unknown"`?